tests/common.py

```diff
@@ -13,20 +13,161 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import codecs
+import collections
 import contextlib
 from io import StringIO
 import os
 import shutil
 import sys
 import tempfile
 import unittest
+import warnings
+from codecs import CodecInfo as CI
```

Comment:
When possible I prefer keeping original names, to avoid confusion (especially when "CI" has other meanings). Would you be OK to use `CodecInfo` at the 4 places where it's used?

Reply:
Done. I only did the `as CI` thing in order to prevent certain lines from being too long. The version of this PR that I just pushed uses `# noqa: E501` instead.